### PR TITLE
openjdk 17

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Need to do:
- [x] Update alias
- [x] Try dropping ARM extra logic (`-arch arm64`, Xcode, etc).
- [ ] Consider testing Linux dependents. Currently skipped and verifying building

### CODE REVIEW

File: Formula/openjdk.rb [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-    # Temporarily use a openjdk 17 preview on Apple Silicon
-    # (because it is better than nothing)
-    url "https://github.com/openjdk/jdk/archive/refs/tags/jdk-17+31.tar.gz"
-    sha256 "9a658a42b2fe3b64ef3b2617395fc8f442f046e43e52b1d3b3a6a9b83d32b2ce"
-    version "16.0.2"
-  else
-    url "https://github.com/openjdk/jdk16u/archive/refs/tags/jdk-16.0.2-ga.tar.gz"
-    sha256 "d1b01bb5e710a973256a11fe852b7e23523ca8ef04997fa29cf459ba5303a476"
-  end
+  url "https://github.com/openjdk/jdk/archive/jdk-17-ga.tar.gz"
```
Comment (cho-m) [inline Formula/openjdk.rb:4]: Currently, versioned repo doesn't have GA tag https://github.com/openjdk/jdk17u/releases. Can consider using that if it is released there.

File: Formula/openjdk.rb [hunk 2]
```diff
@@ -56,8 +47,8 @@ class Openjdk < Formula
   resource "boot-jdk" do
     on_macos do
       if Hardware::CPU.arm?
-        url "https://download.java.net/java/early_access/jdk17/31/GPL/openjdk-17-ea+31_macos-aarch64_bin.tar.gz"
-        sha256 "bf0acde8615ad1bef9c3696128531bec13bcdc3c28baca687bab6902c4b5c7f7"
+        url "https://download.java.net/java/GA/jdk17/0d483333a00540d886896bac774ff48b/35/GPL/openjdk-17_macos-aarch64_bin.tar.gz"
+        sha256 "b5bf6377aabdc935bd72b36c494e178b12186b0e1f4be50f35134daa33bda052"
       else
         url "https://download.java.net/java/GA/jdk15.0.2/0d1cfde4252546c6931946de8db48ee2/7/GPL/openjdk-15.0.2_osx-x64_bin.tar.gz"
         sha256 "578b17748f5a7d111474bc4c9b5a8a06b4a4aa1ba4a4bc3fef014e079ece7c74"
```
Comment (Bo98) [inline Formula/openjdk.rb:54]: Shouldn't boot-jdk should be 16 here (except ARM)?

File: Formula/openjdk.rb (hunk 1 above)
Comment (cho-m) [reply Formula/openjdk.rb:4]: As https://github.com/openjdk/jdk17u/archive/jdk-17-ga.tar.gz was released, I will update to this URL once CI finishes.

**Comment by cho-m** [comment]
It might be possible to drop part of the ARM-specific logic. The `--openjdk-target` and `-arch arm64` seem to be for cross-compilation, which may have been used when boot JDK was still x86_64.

Also, `openjdk` still builds/tests after removing Xcode and framework copying, but I don't know what the exact reason for adding these were.

---

ARM CI:
- [ ] `brew fetch --build-from-source --retry acme`
- [ ] `brew test --retry --verbose dependency-check`
- [ ] `brew install --build-from-source link-grammar`
- [ ] `brew install --build-from-source msc-generator`

Mojave CI:
- [ ] `brew install --build-from-source cbmc`
- [ ] `brew install --build-from-source link-grammar`
- [ ] `brew test --retry --verbose micronaut`
- [ ] `brew install --build-from-source sqoop`
- [ ] `brew install --build-from-source subversion`

**Comment by SMillerDev** [comment]
> Also, openjdk still builds/tests after removing Xcode and framework copying, but I don't know what the exact reason for adding these were.

I believe it linked incorrectly with the DTKs.

**Comment by cho-m** [comment]
I'll give CI a try without the framework copying to see how it looks. If there are any issues, I'll restore the framework logic but probably leave out the cross-compile logic (e.g. `-arch arm64`, `--with-build-jdk`, ...)

**Comment by cho-m** [comment]
Same CI failures as previous run. The formulae directly impacted by update were `micronaut` and `cbmc`. These were both switched to `openjdk@11` in separate PRs.

The remaining failures were previously seen and unrelated to update. These can be fixed outside this PR.

So, should be ready to review and merge.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**